### PR TITLE
Consume a saas application

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following is a first stab at implementing a consume offer
bundle change type. I'm unsure about what the consume changes
requires for the "required" dependency, but I'm sure that'll fallout
soon enough.

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -311,6 +311,13 @@ func (r *resolver) handleRelations(addedApplications map[string]string) {
 
 // handleOffers populates the change set with "CreateOffer" records.
 func (r *resolver) handleOffers(addedApplications map[string]string) {
+	for name, saasSpec := range r.bundle.Saas {
+		r.changes.add(newConsumeOfferChange(ConsumeOfferParams{
+			URL:             saasSpec.URL,
+			ApplicationName: name,
+		}))
```
Comment (SimonRichardson) [inline handlers.go]: @achilleasa any idea what's required here?

**Comment by achilleasa** [comment]
To answer the question re: required field content we can leave it blank as this is an external dependency and the order that this change will be applied does not really matter in this case.

File: handlers.go (hunk 1 above)
Comment (achilleasa) [reply handlers.go]: Perhaps the ApplicationName field is redundant as `juju consume` works fine when provided with just the offer URL. I think it's fine to keep it as we do need it for the Description()/GUIArgs() implementations.

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
https://trello.com/c/exbpHlMa/1579-cmr-update-bundlechanges-to-offer-consume-type

**Pull request merged**